eth/backend.go

```diff
@@ -274,6 +274,22 @@ func New(ctx context.Context, stack *node.Node, config *ethconfig.Config, logger
 	}
 	latestBlockBuiltStore := builder.NewLatestBlockBuiltStore()
 
+	if config.WitnessCachePurge {
+		log.Warn("Purge cache enabled, purging...", "zkevm.witness-cache-purge", config.WitnessCachePurge)
+		tx, err := chainKv.BeginRw(ctx)
+		if err != nil {
+			return nil, err
+		}
+		defer tx.Rollback()
+		hermezDb := hermez_db.NewHermezDb(tx)
+		if err := hermezDb.PurgeWitnessCaches(); err != nil {
+			return nil, fmt.Errorf("failed to purge witness caches: %w", err)
+		}
+		if err = tx.Commit(); err != nil {
+			return nil, fmt.Errorf("tx.Commit: %w", err)
+		}
+	}
+
```

Comment:
It might be worth looking at the pre-start tasks for this, we use these already for rebuilding the datastream.  It helped with managing the DB lifecycle a lot as we have tighter control there on the opening and closing of RW transactions as the scope is always smaller.  Refactors in backend.go could cause you some headaches if you're opening and committing transactions here.

Example:
![image](https://github.com/user-attachments/assets/8b6deb22-38d6-4f26-a9dc-aadfe96a1f03)